### ISSUE TITLE
Auth.Write throw exception with roleid/secretid

### ISSUE THREAD

**Issue by yongkeecho** [Chatham/Vault.NET#23]
The code throw System.Collections.ObjectModel.ReadOnlyCollection`1[System.Exception]. Anything wrong in my code?

private static async Task<string> GetToken(VaultClient vaultClient)
 {
            var loginRequest = new LoginRequest
            {
                RoleId = "***",
                SecretId = "***"
            };
            var loginResponse = await vaultClient.Auth.Write<LoginRequest, NoData>("auth/approle/login", loginRequest);
            return loginResponse.Auth.ClientToken;
}

**Comment by chrishoffman** [comment]
I created #24 that tests out the approle backend.  Without more information, I am not sure the issue you are running into.

**Comment by yongkeecho** [comment]
In CLI, it works fine
$ vault write auth/approle/login role_id=ab1c859... secret_id=b4fcf5d8...
Key                     Value
---                     -----
token                   b80dac78-1477-333f-06cd-3b587a17b0de
token_accessor          827bdd37-2e3a-a4c0-3e95-8936dc4a0d19
token_duration          768h0m0s
token_renewable         true
token_policies          [approle/ia-dev-team/role default]
token_meta_role_name    "ia-dev-team"

In the code above, it throws aggregation exception in vault.Auth.Write method and its inner exception message is  System.Collections.ObjectModel.ReadOnlyCollection`1[System.Exception].

**Comment by chrishoffman** [comment]
The issue is the you are using the auth prefix in your api path with the auth part of the client. Just update your api path to “approle/login”.

**Comment by yongkeecho** [comment]
Thank you Chris.
I could login with approle. Now I have another issue.

I set the vault client token the approle token.
vaultClient.Token = loginResponse.Auth.ClientToken;

But when I tried to set a secret, I had unknown exception. Reading a secret also throw exception. The code worked fine with userpass login.

Console.WriteLine(SetGetSecret(vaultClient, "secret/team/app", "key1", "vaule1");

private static async Task<string> SetGetSecret(IVaultClient vaultClient, string path, string key, string value)
{
            try
            {
                var data = new Dictionary<string, string> {{key, value}};
                await vaultClient.Secret.Write(path, data);
                var secret = await vaultClient.Secret.Read<Dictionary<string, string>>(path);
                return secret.Data[key];
            }
            catch (Exception exception)
            {
                Console.WriteLine(exception.ToString());
            }
            return null;
}

It works fine in CLI with approle credential though.
$ vault write secret/team/app/key1 vaule="value1"
Key       Value
------ ---------
key1      value1
